monai/apps/pathology/transforms/post/array.py

```diff
@@ -0,0 +1,336 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Sequence, Union
+
+import numpy as np
+import torch
+
+from monai.config.type_definitions import NdarrayOrTensor
+from monai.data.meta_tensor import get_track_meta
+from monai.transforms.croppad.array import BoundingRect
+
+from monai.transforms.post.array import GetInstanceLevelSegMap, Activations, AsDiscrete
+from monai.transforms.transform import Transform
+from monai.utils import convert_to_numpy, convert_to_tensor, optional_import
+from monai.utils.enums import TransformBackends
+
+find_contours, _ = optional_import("skimage.measure", name="find_contours")
+moments, _ = optional_import("skimage.measure", name="moments")
+
+
+__all__ = ["PostProcessHoVerNetOutput"]
+
+
+def _coords_to_pixel(current, previous) -> Sequence[int]:
+    """
+    For contour coordinate generation.
+    Given the previous and current border positions,
+    returns the int pixel that marks the extremity
+    of the segmented pixels
+
+    Args:
+        current: coordinates of the current border positions
+        previous: coordinates of the previous border positions
+    """
+
+    p_delta = (current[0] - previous[0], current[1] - previous[1])
+
+    if p_delta == (0.0, 1.0) or p_delta == (0.5, 0.5) or p_delta == (1.0, 0.0):
+        row = int(current[0] + 0.5)
+        col = int(current[1])
+    elif p_delta == (0.0, -1.0) or p_delta == (0.5, -0.5):
+        row = int(current[0])
+        col = int(current[1])
+    elif p_delta == (-1, 0.0) or p_delta == (-0.5, -0.5):
+        row = int(current[0])
+        col = int(current[1] + 0.5)
+    elif p_delta == (-0.5, 0.5):
+        row = int(current[0] + 0.5)
+        col = int(current[1] + 0.5)
+
+    return row, col
+
+
+def _dist_from_topleft(sequence, h, w) -> int:
+    """
+    For contour coordinate generation.
+    Each sequence of cordinates describes a boundary between
+    foreground and background starting and ending at two sides
+    of the bounding box. To order the sequences correctly,
+    we compute the distannce from the topleft of the bounding box
+    around the perimeter in a clock-wise direction.
+
+    Args:
+        sequence: list of border points
+        h: height of the bounding box
+        w: width of the bounding box
+
+    Returns:
+        the distance round the perimeter of the bounding
+            box from the top-left origin
+    """
+
+    first = sequence[0]
+    if first[0] == 0:
+        distance = first[1]
+    elif first[1] == w - 1:
+        distance = w + first[0]
+    elif first[0] == h - 1:
+        distance = 2 * w + h - first[1]
+    else:
+        distance = 2 * (w + h) - first[0]
+
+    return distance
+
+
+def _sp_contours_to_cv(contours, h, w) -> np.array:
```

Comment:
Doesn't make sense to create a new components for this? Can it be useful in other applications?

Reply:
I was going to add this in another pr, but now I've created it in this [new commit](https://github.com/Project-MONAI/MONAI/pull/5186/commits/b253302a7a58e1ae73826e1de68eb0153d9c56cb) in this pr. Thanks!